node/heartbeat/sender.go

```diff
@@ -58,8 +64,10 @@ func NewSender(
 func (s *Sender) SendHeartbeat() error {
 
 	hb := &Heartbeat{
-		Payload: []byte(fmt.Sprintf("%v", time.Now())),
-		ShardID: s.shardCoordinator.SelfId(),
+		Payload:         []byte(fmt.Sprintf("%v", time.Now())),
```

Comment:
This should be better done in the receive part, in `ProcessReceivedMessage`. Every node should have a representation of when a peer sends a hartbeat from their perspective, not from the sender perspective. Even though this means that you would get different results questioning different nodes for a given peer, it's more correct in a decentralised setting. Consider that I can send any timestamp I want here (past or future) and every node will accept this. The equivalent in a centralised setting is sending a hartbeat to a server (like we have in our cryptobubbles game), the player sends a ping to the server and the server notarises the timestamp relative to itself.

Reply:
Partially agree. I would leave this field on, and add a new one (computed=time.Now()) in the heartbeat structure. The difference between the computed one and the received one (if positive) will represent the node's latency. If the difference is negative, an output of 0ms latency can be done.

Reply:
Corco, as we discussed, there is no need to remove it because that pay load represents the message which is signed by the sender when sending a heartbeat in order to authenticate him when another peer receives a heartbeat.
However, we can use Iulian's idea

Reply:
👍